classes/plugins/ThemePlugin.inc.php

```diff
@@ -59,18 +76,40 @@ function register($category, $path) {
 			return true;
 		}
 
-		// Fire an initialization method which themes should use to add
-		// styles, scripts and fonts
-		$this->init();
+		// Themes must initialize their functionality after all theme plugins
+		// have been loaded in order to make use of parent/child theme
+		// relationships
+		HookRegistry::register('PluginRegistry::categoryLoaded::themes', array($this, 'themeRegistered'));
+		HookRegistry::register('PluginRegistry::categoryLoaded::themes', array($this, 'initAfter'));
 
-		$this->_registerTemplates();
-		$this->_registerStyles();
-		$this->_registerScripts();
+		// Save any theme options displayed on the appearance and site settings
+		// forms
+		HookRegistry::register('appearanceform::execute', array($this, 'saveOptionsForm'));
+		HookRegistry::register('appearanceform::readuservars', array($this, 'readOptionsFormUserVars'));
+		HookRegistry::register('sitesetupform::execute', array($this, 'saveOptionsForm'));
+		HookRegistry::register('sitesetupform::readuservars', array($this, 'readOptionsFormUserVars'));
 
 		return true;
 	}
 
 	/**
+	 * Fire the init() method when a theme is registered
```

Comment:
Yeah, it's just a wrapper to keep the `init` method (where themes should do all their work) clean from any "mucking about" we might do with the load procedure later.